src/AstGen.zig

```diff
@@ -551,7 +551,18 @@ fn expr(gz: *GenZir, scope: *Scope, rl: ResultLoc, node: ast.Node.Index) InnerEr
         .mul_wrap => return simpleBinOp(gz, scope, rl, node, .mulwrap),
         .div      => return simpleBinOp(gz, scope, rl, node, .div),
         .mod      => return simpleBinOp(gz, scope, rl, node, .mod_rem),
-        .bit_and  => return simpleBinOp(gz, scope, rl, node, .bit_and),
+        .bit_and  => {
+            const current_ampersand_token = main_tokens[node];
+            if (token_tags[current_ampersand_token + 1] == .ampersand) {
```

Comment:
I'm not sure this fixes the error. Does 
```zig
const y = &&ident;
```
Work correctly after this?
If so, please add a test.
The way I would do this is to see if the previous token was an identifier (and maybe the one after) in addition to what you have, because then that would be invalid.

Reply:
Added the test for the case you have mentioned.

My understanding is that `bit_and`, as a binary operator, guarantees that the left operand is always an expression, i.e. identifier, literal boolean etc. So we should be okay when checking only for the right-hand side of the expression.